src/storage/src/hummock/file_cache/store.rs

```diff
@@ -12,431 +12,533 @@
-
-        for buffer in self.buffers {
-            let blen = buffer.len();
-
-            let timer = self.store.metrics.disk_write_latency.start_timer();
-            let offset = self.store.cache_file.append(buffer).await? / self.block_size as u64;
-            timer.observe_duration();
-
-            if boff.is_none() {
-                boff = Some(offset.try_into().unwrap());
-            }
-
-            self.store.metrics.disk_write_bytes.inc_by(blen as f64);
-            self.store.metrics.disk_write_io_size.observe(blen as f64);
-        }
-
-        let boff = boff.unwrap();
+impl Value for Box<Block> {
+    fn serialized_len(&self) -> usize {
+        self.raw_data().len()
+    }
 
-        for bloc in &mut self.blocs {
-            bloc.bidx += boff;
-        }
+    fn write(&self, mut buf: &mut [u8]) {
+        buf.put_slice(self.raw_data())
+    }
 
-        // Write guard is only needed when updating meta file, for data file is append-only.
-        let mut guard = self
-            .store
-            .meta_file
-            .write()
-            .instrument(tracing::trace_span!("meta_write_lock_update_slots"))
-            .await;
+    fn read(buf: &[u8]) -> Self {
+        let data = Bytes::copy_from_slice(buf);
+        let block = Block::decode_from_raw(data);
+        Box::new(block)
+    }
+}
 
-        for (key, bloc) in self.keys.iter().zip_eq_fast(self.blocs.iter()) {
-            slots.push(guard.insert(key, bloc)?);
-        }
+impl Value for Box<Sstable> {
+    fn weight(&self) -> usize {
+        std::mem::size_of::<Self>()
+    }
 
-        Ok((self.keys, slots))
+    fn serialized_len(&self) -> usize {
+        8 + self.meta.encoded_size() // id (8B) + meta size
     }
-}
 
-pub struct StoreOptions {
-    pub dir: String,
-    pub capacity: usize,
-    pub buffer_capacity: usize,
-    pub cache_file_fallocate_unit: usize,
-    pub cache_meta_fallocate_unit: usize,
-    pub cache_file_max_write_size: usize,
+    fn write(&self, mut buf: &mut [u8]) {
+        buf.put_u64(self.id);
+        // TODO(MrCroxx): avoid buffer copy
+        let mut buffer = vec![];
+        self.meta.encode_to(&mut buffer);
+        buf.put_slice(&buffer[..])
+    }
 
-    pub metrics: FileCacheMetricsRef,
+    fn read(mut buf: &[u8]) -> Self {
+        let id = buf.get_u64();
+        let meta = SstableMeta::decode(&mut buf).unwrap();
+        Box::new(Sstable::new(id, meta))
+    }
 }
 
-pub struct Store<K, V>
+#[derive(Clone)]
+pub enum FileCache<K, V>
 where
-    K: TieredCacheKey,
-    V: TieredCacheValue,
+    K: Key + Copy,
+    V: Value,
 {
-    dir: String,
-    _capacity: usize,
-
-    fs_type: FsType,
-    _fs_block_size: usize,
-    block_size: usize,
-    buffer_capacity: usize,
-    cache_file_max_write_size: usize,
-
-    meta_file: Arc<AsyncRwLock<MetaFile<K>>>,
-    cache_file: CacheFile,
-
-    freelist: Arc<RwLock<Vec<SlotId>>>,
-
-    metrics: FileCacheMetricsRef,
-
-    _phantom: PhantomData<V>,
+    None,
+    Foyer {
```

Comment:
May add doc to describe the difference between `Foyer` and `FoyerRuntime`.

Reply:
`Foyer` is removed and the original `FoyerRuntime` is renamed to `Foyer` now.